libp2p-swarm/src/transport.rs

```diff
@@ -384,6 +383,37 @@ where
 	}
 }
 
+/// > **Note**: This type is needed because of the lack of `-> impl Trait` in Rust. It can be
```

Comment:
Is that actually true? It's an enum and I don't think `impl Trait` allows for anonymous sum types.

Reply:
If Rust had `impl Trait` we could use the `Either` enum from the `futures` crate and add some modifiers to it. This custom enum is a combination of `Either` and these modifiers.

Reply:
Maybe say that here then because as it stands this note isn't much more than a `TODO` and doesn't have any place in the docs.